openmc/weight_windows.py

```diff
@@ -420,6 +422,8 @@ def from_hdf5(cls, group, meshes):
         upper_ww_bounds = group['upper_ww_bounds'][()]
         survival_ratio = group['survival_ratio'][()]
 
+        mesh = meshes[mesh_id]
```

Comment:
perhaps this from_hdf5 class method does not need to request a dictionary of mesh ids and mesh objects. As we could get the mesh id from the mesh.id attribute. Might be easier for users to pass in a list of meshes instead of the dict

Reply:
As in many of our other features, the mesh being used is identified by its ID in the HDF5 file. Were we to pass a list of mesh objects here, my feeling is that the code would become clunky when checking whether or not the mesh used with the weight windows object is present. We'd have to search over the list of meshes, checking whether or not there is an ID match.

Reply:
Going to resolve this @shimwell, but if this is a structural change you feel strongly about, please raise an issue so we can discuss it further!